pandas/tests/frame/test_apply.py

```diff
@@ -49,7 +49,8 @@ def test_apply(self, float_frame):
 
         # invalid axis
         df = DataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]], index=["a", "a", "c"])
-        with pytest.raises(ValueError):
+        msg = "No axis named 2 for object type <class 'pandas.core.frame.DataFrame'>"
```

Comment:
I find this a bit confusing error message. Can we directly try to improve it as well? 
First, we can use the short "DataFrame" instead of the long "<class 'pandas.core.frame.DataFrame'>. Second, we could indicate that for DataFrame the axis can only be 0 or 1

Reply:
I 100% agree with your assessments on the error messages themselves. I also found that some of them are different across operating systems, which seems even more problematic. However, is that within the purview of this issue or should another one be made?

Reply:
For me, that's perfectly fine to do in this PR (it is here we are testing and thus looking at error messages, so let's try to directly improve them to the extent possible)